docs/pages/guides/testing-with-jest.md

```diff
@@ -47,6 +51,17 @@ We would like to point out [`transformIgnorePatterns`](https://jestjs.io/docs/co
 }
 ```
 
+By default, Jest runs the code as JavaScript and only supports Node.js based syntax. If your project uses syntax such as importing files with **.jsx** or **.tsx** extensions, use [`transform`](https://jestjs.io/docs/code-transformation) configuration option and explicitly define `babel-jest` as the transformer.
```

Comment:
Are you sure this is required?

Reply:
I manually tested with an example of using `.jsx` file. It does throw the error when running tests with jest. Here is an example: 

![CleanShot 2022-07-29 at 19 50 00@2x](https://user-images.githubusercontent.com/10234615/181780255-2b10e49c-0ded-45de-bcbb-d73cfedec01d.jpg)

Reply:
In the above, the `TextComponent` is saved inside a `TextComponent.jsx` file.

Reply:
And you are using the jest-expo preset?!

Reply:
Yep:

![CleanShot 2022-07-29 at 19 54 00@2x](https://user-images.githubusercontent.com/10234615/181781026-9699047f-44a0-4ac5-92cf-44299698c97b.jpg)

Reply:
If so, I think this is a bug in the preset, not something we should have to document separately. I suspect this might be the reason:
![image](https://user-images.githubusercontent.com/852069/181781114-e7f60643-aaa9-4cdf-9ab2-601d02ebc80a.png)

We add something to `transform` in our preset, without including the default.

Reply:
Reading this line of the code: https://github.com/expo/expo/blob/master/packages/jest-expo/jest-preset.js#L10, you will find that if the `transform` key is not present, it is being set to an empty object.

Reply:
Yes, and https://github.com/expo/expo/blob/master/packages/jest-expo/jest-preset.js#L22-L24 sets it only for the asset pattern, so that is exactly the case the docs describe. We need to include the default `"\\.[jt]sx?$": "babel-jest"` there.

Reply:
In that case, I don't think we should update the docs.

Reply:
Right, we need to fix the `jest-expo` preset, so it works out of the box, and we don't need this addition to the docs 👍🏻